### PR TITLE
Switching to GitHub actions

### LINKED ISSUE #2: Update to Node 20 (current LTS)
## Description
Upgrading the project's Node.js dependency to version 20 (current LTS) will ensure that our software remains supported with the latest security patches and performance improvements. Node.js 20 introduces several new features and optimizations that can enhance application performance and developer productivity. This upgrade will also ensure compatibility with modern libraries and frameworks that require the latest Node.js features.

## Acceptance Criteria

- package.json Engines requires Node 20, npm 10
- Ensure Node 20 build and test works locally
- Update any CI/CD to test Node 20

## Use Case

When developers are implementing new features or maintaining existing code, they should have access to the latest JavaScript features, improved diagnostics, and enhanced performance. Upgrading to Node.js 20 will minimize potential security vulnerabilities and compatibility issues with other tools and libraries that depend on newer Node.js versions. This will streamline development workflows and reduce the time spent on troubleshooting and technical debt.

### LINKED ISSUE #3: Replace Buddy with GitHub Actions
## Description

Upgrading the project from Buddy to GitHub actions will ensure parity with other Alley projects, reduce costs, and allow leveraging 'shared' common workflows or actions.

## Acceptance Criteria

- All CI/CD pipelines (eg. PR tests or deployment) are replaced by equivalent GitHub actions
- Any secrets are recreated or moved to GitHub secrets (environment or otherwise)
- Buddy disabled for the project

### PR DESCRIPTION
**Pull request by srtfisher**
Resolves #2. Fixes #3.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+          ssh-keyscan ${{ env.SERVER_IP }} >> ~/.ssh/known_hosts
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.2'
```
Comment (renatonascalves) [inline .github/workflows/deploy.yml]: not PHP 8.3? The other workflow uses PHP 8.3

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            --exclude .git \
+            --exclude tests \
+            --exclude storage/debugbar \
+            --exclude .env
+
+      - name: Execute remote commands
+        run: |
+          ssh ${{ env.SERVER_USER }}@${{ env.SERVER_IP }} "cd ${{ env.DEPLOY_DIR }} && \
+          export PATH=/usr/local/bin:\$PATH && \
+          php8.3 artisan migrate --force && \
```
Comment (renatonascalves) [inline .github/workflows/deploy.yml:59]: Here it seems to use php 8.3 too.

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (srtfisher) [reply .github/workflows/deploy.yml]: Good eye!

**Review by renatonascalves — APPROVED** [review]
🌵

**Pull request merged**